tabs/download/download.py

```diff
@@ -72,9 +73,29 @@ def search_models(name):
 
 
 def fetch_pretrained_data():
-    response = requests.get(json_url)
-    response.raise_for_status()
-    return response.json()
+    pretraineds_custom_path = os.path.join(
+        "rvc", "models", "pretraineds", "pretraineds_custom"
+    )
+    os.makedirs(pretraineds_custom_path, exist_ok=True)
+    try:
+        with open(os.path.join(pretraineds_custom_path, json_url.split("/")[-1]), 'r') as f:
+            data=json.load(f)
+    except:
+        try:
+            response = requests.get(json_url)
+            response.raise_for_status()
+            data = response.json()
+            with open(os.path.join(pretraineds_custom_path, json_url.split("/")[-1]), 'w', encoding="utf-8") as f:
+                json.dump(
+                    data,
+                    f,
+                    indent=2,
+                    separators=(",", ": "),
+                    ensure_ascii=False,
+                )
+        except:
+            data = {"Titan": {"32k": {"D": "null", "G": "null"},},}
```

Comment:
why not return the current list of pretraineds, even if it becomes outdated in the future, better than null

Reply:
> why not return the current list of pretraineds, even if it becomes outdated in the future, better than null

because it is just a list of pretraineds to download from the Download tab. You dont need to download what has already been downloaded.